core/src/components/text/text.ionic.scss

```diff
@@ -0,0 +1,247 @@
+/*
+    All the CSS for typography is added on the text.scss temporarily because the correct architecture for the
+    new them implementation aren't ready.
+
+    In the future, all these sass variable and css created needs to be putted at theme level in terms of architecture.
+
+    The typography of an app, needs to be spitted in ios, md and ionic themes.
+*/
+
+// Typography
+// --------------------------------------------------
+/// Font Family
+$os-font-family: "Inter", sans-serif;
+
+/// Font sizes
+/// @prop - Font size small
+$os-font-size-s: 12px;
+
+/// @prop - Font size base
+$os-font-size-base: 14px;
+
+/// @prop - Font size large
+$os-font-size-l: 16px;
+
+/// Font size displays
+/// @prop - Display font size small
+$os-display-s: 32px;
+
+/// @prop - Display font size medium
+$os-display-m: 36px;
+
+/// Headings
+/// @prop - Font size of heading level 1
+$os-h1-font-size: dynamic-font(28px) !default;
+
+/// @prop - Font size of heading level 2
+$os-h2-font-size: dynamic-font(26px) !default;
+
+/// @prop - Font size of heading level 3
+$os-h3-font-size: dynamic-font(24px) !default;
+
+/// @prop - Font size of heading level 4
+$os-h4-font-size: dynamic-font(22px) !default;
+
+/// @prop - Font size of heading level 5
+$os-h5-font-size: dynamic-font(20px) !default;
+
+/// @prop - Font size of heading level 6
+$os-h6-font-size: dynamic-font(18px) !default;
+
+/// Font Weights
+/// @prop - Font weight light
+$os-font-light: 300;
+
+/// @prop - Font weight regular
+$os-font-regular: 400;
+
+/// @prop - Font weight medium
+$os-font-medium: 500;
+
+/// @prop - Font weight semibold
+$os-font-semibold: 600;
+
+/// @prop - Font weight bold
+$os-font-bold: 700;
+
+/// Line heights
+/// @prop - Line height size xsmall
+$os-line-height-xs: 16px;
+
+/// @prop - Line height size small
+$os-line-height-s: 20px;
+
+/// @prop - Line height size medium
+$os-line-height-m: 24px;
+
+/// @prop - Line height size large
+$os-line-height-l: 28px;
+
+/// @prop - Line height size xlarge
+$os-line-height-xl: 32px;
+
+/// @prop - Line height size xxlarge
+$os-line-height-xxl: 36px;
+
+/// @prop - Display line height small
+$os-line-height-display-s: 44px;
+
+/// @prop - Display line height medium
+$os-line-height-display-m: 48px;
+
+/// Letter spacing
+/// @prop - Letter spacing none
+$os-letter-spacing-none: 0em;
+
+/// @prop - Letter spacing small
+$os-letter-spacing-s: 0.1em;
+
+/// @prop - Letter spacing medium
+$os-letter-spacing-m: 0.15em;
+
+html {
```

Comment:
Many of these selectors have no effect in this stylesheet. With shadow dom encapsulation, these styles only apply to elements that are within the shadow-root. 

Since these elements are slotted, they would need to use `::slotted(h1)`, etc. to target them. 

However, based on the implementation pattern here I don't think this is the path you are wanting to take. 

These styles should be global styles, not component-level styles. 

For example, a stylesheet like `ionic.global.md.scss`, but specifically for the ionic theme.

Reply:
On the `scss` file I have the explanation for this, however, I will create a separated `scss` file under the CSS folder to contain temporarily all the global css, as we discuss.